internal/cadence/lint.go

```diff
@@ -0,0 +1,215 @@
+		Args:    cobra.MinimumNArgs(1),
+	},
+	Flags: &lintFlags,
+	RunS:  lint,
+}
+
+type severity string
+
+const (
+	errorSeverity   severity = "error"
+	warningSeverity severity = "warning"
+)
+
+func lint(
+	args []string,
+	globalFlags command.GlobalFlags,
+	logger output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (command.Result, error) {
+	filePaths := args
+	result, err := lintFiles(state, filePaths...)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func lintFiles(
+	state *flowkit.State,
+	filePaths ...string,
+) (
+	*lintResult,
+	error,
+) {
+	l := newLinter(state)
+	results := make([]fileResult, 0)
+	exitCode := 0
+
+	for _, location := range filePaths {
+		diagnostics, err := l.lintFile(location)
+		if err != nil {
+			return nil, err
+		}
+
+		// Sort for consistent output
+		sortDiagnostics(diagnostics)
+		results = append(results, fileResult{
+			FilePath:    location,
+			Diagnostics: diagnostics,
+		})
+
+		// Set the exitCode to 1 if any of the diagnostics are error-level
+		// In the future, this may be configurable
+		for _, diagnostic := range diagnostics {
+			severity := getDiagnosticSeverity(diagnostic)
+			if severity == errorSeverity {
+				exitCode = 1
+				break
+			}
+		}
+	}
+
+	return &lintResult{
+		Results:  results,
+		exitCode: exitCode,
+	}, nil
+}
+
+func getDiagnosticSeverity(
+	diagnostic analysis.Diagnostic,
+) severity {
+	if isErrorDiagnostic(diagnostic) {
+		return errorSeverity
+	}
+	return warningSeverity
+}
+
+// Sort diagnostics in order of precedence: start pos -> category -> message
+func sortDiagnostics(
+	diagnostics []analysis.Diagnostic,
+) {
+	slices.SortFunc(diagnostics, func(a analysis.Diagnostic, b analysis.Diagnostic) int {
+		if r := a.Range.StartPos.Offset - b.Range.StartPos.Offset; r != 0 {
+			return r
+		}
+
+		if r := strings.Compare(a.Category, b.Category); r != 0 {
+			return r
+		}
+
+		return strings.Compare(a.Message, b.Message)
+	})
+}
+
+func renderDiagnostic(diagnostic analysis.Diagnostic) string {
+	categoryColor := aurora.RedFg
+	if getDiagnosticSeverity(diagnostic) == warningSeverity {
+		categoryColor = aurora.YellowFg
+	}
+
+	startPos := diagnostic.Range.StartPos
+	locationText := fmt.Sprintf("%s:%d:%d:", diagnostic.Location.String(), startPos.Line, startPos.Column)
+	categoryText := fmt.Sprintf("%s:", diagnostic.Category)
+
+	return fmt.Sprintf("%s %s %s",
+		aurora.Gray(12, locationText).String(),
+		aurora.Colorize(categoryText, categoryColor).String(),
+		diagnostic.Message,
+	)
+}
+
+func (r *lintResult) String() string {
+	var sb strings.Builder
+	var numProblems int
+
+	for _, result := range r.Results {
+		if len(result.Diagnostics) == 0 {
+			continue
+		}
+
+		numProblems += len(result.Diagnostics)
+
+		for _, diagnostic := range result.Diagnostics {
+			sb.WriteString(fmt.Sprintf("%s\n\n", renderDiagnostic(diagnostic)))
+		}
+	}
+
+	if numProblems == 0 {
+		sb.WriteString("No problems found")
```

Comment:
Should this be green and say "All __ passing" or something more positive than "no problems."

Reply:
I just removed it entirely.  This seems to be more in line with what other tools do, lmk what you think